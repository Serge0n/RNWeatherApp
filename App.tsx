import {
  DarkTheme,
  DefaultTheme,
  NavigationContainer,
} from '@react-navigation/native'
import { NavStack } from 'navigation'
import { StatusBar, useColorScheme } from 'react-native'
import RNBootSplash from 'react-native-bootsplash'
import { SafeAreaProvider } from 'react-native-safe-area-context'

export function App(): JSX.Element {
  const isDarkMode = useColorScheme() === 'dark'

  const hideSplash = () => RNBootSplash.hide({ fade: true })

  return (
    <SafeAreaProvider>
      <NavigationContainer
        onReady={hideSplash}
        theme={isDarkMode ? DarkTheme : DefaultTheme}
      >
        <StatusBar />
        <NavStack />
      </NavigationContainer>
    </SafeAreaProvider>
  )
}
