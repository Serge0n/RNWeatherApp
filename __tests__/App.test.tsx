import { describe, expect, it, jest } from '@jest/globals'
import { render, screen } from '@testing-library/react-native'
import 'react-native'
import { hide } from 'react-native-bootsplash'
import { App } from '../App'

const mockedUseColorScheme = jest.fn()
jest.mock('react-native/Libraries/Utilities/useColorScheme', () => ({
  default: mockedUseColorScheme,
}))

describe('App', () => {
  it('hides Splash Screen ', () => {
    render(<App />)
    expect(hide).toBeCalledTimes(1)
  })
  it('renders correctly with dark theme', () => {
    mockedUseColorScheme.mockReturnValue('dark')
    render(<App />).toJSON()
    expect(screen).toMatchSnapshot()
  })
  it('renders correctly with light theme', () => {
    mockedUseColorScheme.mockReturnValue('light')
    render(<App />).toJSON()
    expect(screen).toMatchSnapshot()
  })
})
