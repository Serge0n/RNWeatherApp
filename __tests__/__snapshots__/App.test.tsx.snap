// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly with dark theme 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(18, 18, 18)"
        color="rgb(10, 132, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(229, 229, 231)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(1, 1, 1)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View>
              <TextInput
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter the city"
                style={
                  [
                    {
                      "borderColor": "rgb(39, 39, 41)",
                      "borderRadius": 8,
                      "borderWidth": 3,
                      "color": "rgb(229, 229, 231)",
                      "fontSize": 24,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                    {
                      "marginHorizontal": 16,
                    },
                  ]
                }
              />
            </View>
            <RCTScrollView
              ListHeaderComponent={<View />}
              data={[]}
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              style={
                {
                  "marginTop": 16,
                  "paddingHorizontal": 16,
                }
              }
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  collapsable={false}
                  onLayout={[Function]}
                >
                  <View />
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</View>
`;

exports[`App renders correctly with light theme 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <View>
              <TextInput
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Enter the city"
                style={
                  [
                    {
                      "borderColor": "rgb(216, 216, 216)",
                      "borderRadius": 8,
                      "borderWidth": 3,
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 24,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                    {
                      "marginHorizontal": 16,
                    },
                  ]
                }
              />
            </View>
            <RCTScrollView
              ListHeaderComponent={<View />}
              data={[]}
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              style={
                {
                  "marginTop": 16,
                  "paddingHorizontal": 16,
                }
              }
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  collapsable={false}
                  onLayout={[Function]}
                >
                  <View />
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</View>
`;
